transformer_engine/jax/sharding.py

```diff
@@ -24,6 +37,81 @@ def _get_mesh_info(resource: str):
     return mesh.shape[resource], resource
 
 
+def get_sharding_map_logic_axis_to_mesh_axis():
+    """
+    Generate a dict to map logical axes to mesh axes.
+    """
+    gsr = global_mesh_resource()
+
+    IS_FSDP_OUTER = bool(int(os.environ.get("NVTE_OUTER_BATCH_FSDP_DIM", False)))
```

Comment:
This is a env-var to specify how to shard tensor along batch-dim for DP + FSDP. If it is `true`, the we expect the batch of  tensors be sharded like `('FSDP', 'DP')`, otherwise `('DP', 'FSDP')`

Reply:
Internal variable also

Reply:
How an env variable can be internal?
End user will never want to use it?

Reply:
This flag should be share across all modules for both `TransformerLayer` and `MultiheadAttention` of  Flax/Praxis. Therefore I put it in `shardind.py` to make is as a common variable instead of one specific to Flax/Praxis.

It is for experimentation at this moment, just in case there are some training configs required to shard their batch along DP first then FSDP, and they would have a var to determine that.